File: tests/rosetta/transpiler/Java/animate-a-pendulum.java
public class Main {
    static double PI = 3.141592653589793;
    static double L = 10.0;
    static double G = 9.81;
    static double dt = 0.2;
    static double phi0 = PI / 4.0;
    static double omega = sqrtApprox(G / L);
    static double t = 0.0;

    static double sinApprox(double x) {
        double term = x;
        double sum = x;
        int n = 1;
        while (n <= 10) {
            double denom = ((Number)(((2 * n) * (2 * n + 1)))).doubleValue();
            term = -term * x * x / denom;
            sum = sum + term;
            n = n + 1;
        }
        return sum;
    }

    static double cosApprox(double x) {
        double term = 1.0;
        double sum = 1.0;
        int n = 1;
        while (n <= 10) {
            double denom = ((Number)(((2 * n - 1) * (2 * n)))).doubleValue();
            term = -term * x * x / denom;
            sum = sum + term;
            n = n + 1;
        }
        return sum;
    }

    static double sqrtApprox(double x) {
        double guess = x;
        int i = 0;
        while (i < 10) {
            guess = (guess + x / guess) / 2.0;
            i = i + 1;
        }
        return guess;
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            for (int step = 0; step < 10; step++) {
                double phi = phi0 * cosApprox(omega * t);
                int pos = ((Number)((10.0 * sinApprox(phi) + 0.5))).intValue();
                System.out.println(String.valueOf(pos));
                t = t + dt;
            }
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        return rt.totalMemory() - rt.freeMemory();
    }
}

File: tests/rosetta/transpiler/Kotlin/animate-a-pendulum.kt
val PI: Double = 3.141592653589793
val L: Double = 10.0
val G: Double = 9.81
val dt: Double = 0.2
val phi0: Double = PI / 4.0
val omega: Double = sqrtApprox(G / L)
var t: Double = 0.0
fun sinApprox(x: Double): Double {
    var term: Double = x
    var sum: Double = x
    var n: Int = 1
    while (n <= 10) {
        val denom: Double = (2 * n) * ((2 * n) + 1).toDouble()
        term = (((0.0 - term) * x) * x) / denom
        sum = sum + term
        n = n + 1
    }
    return sum
}

fun cosApprox(x: Double): Double {
    var term: Double = 1.0
    var sum: Double = 1.0
    var n: Int = 1
    while (n <= 10) {
        val denom: Double = ((2 * n) - 1) * (2 * n).toDouble()
        term = (((0.0 - term) * x) * x) / denom
        sum = sum + term
        n = n + 1
    }
    return sum
}

fun sqrtApprox(x: Double): Double {
    var guess: Double = x
    var i: Int = 0
    while (i < 10) {
        guess = (guess + (x / guess)) / 2.0
        i = i + 1
    }
    return guess
}

fun main() {
    for (step in 0 until 10) {
        val phi: Double = phi0 * cosApprox(omega * t)
        val pos: Int = (10.0 * sinApprox(phi)) + 0.5.toInt()
        println(pos.toString())
        t = t + dt
    }
}
